service/src/main/java/uk/nhs/adaptors/gp2gp/common/configuration/AppInitializer.java

```diff
@@ -0,0 +1,26 @@
+package uk.nhs.adaptors.gp2gp.common.configuration;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component(value = "appInitializer")
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AppInitializer implements InitializingBean {
+    private final AppConfiguration appConfiguration;
+    private final CustomTrustStore customTrustStore;
+
+    @Override
+    public void afterPropertiesSet() {
+        LOGGER.info("Running app initializer");
+        if (StringUtils.isNotBlank(appConfiguration.getTrustStoreUrl())) {
```

Comment:
Should password be checked?

Reply:
It is checked when connections is attempted. Unless of course there is a required password regex.

Reply:
Could maybe improve error handling by taking this approach so the user knows exactly whats wrong with a helpful message

Reply:
The problem is that if we do not have trust store url we are supposed to run service without trust store. What do you think about a warning message mentioning that we are running service without the trust store?

Reply:
That's a good idea! Maybe also if the url is set then an additional check and error message for password? (Assuming password cannot be blank)

Reply:
The problem could arise if we want to test a trust store which does not contain a password. I think it is fine like that. password is being validated during the connection process anyway.